backend/src/main/java/ca/bc/gov/app/validator/SubmissionValidatorService.java

```diff
@@ -68,6 +74,51 @@ public Mono<ClientSubmissionDto> validate(
     return validateSubmissionDto(request).switchIfEmpty(validateSubmissionData(request, source));
   }
 
+  public Mono<Void> validateSubmissionLimit(JwtAuthenticationToken principal) {
+    String userId = JwtPrincipalUtil.getUserId(principal);
+    return checkSubmissionLimit(userId);
+  }
+
+  public Mono<Void> checkSubmissionLimit(String userId) {
+    String provider = extractPrefix(userId);
+    LocalDateTime endTime = LocalDateTime.now();
+    LocalDateTime startTime;
+    long maxSubmissions;
+
+    if ("IDIR".equalsIgnoreCase(provider)) {
+      startTime = endTime.minus(configuration.getIdirSubmissionTimeWindow());
+      maxSubmissions = configuration.getIdirMaxSubmissions();
+    } 
+    else if ("BCSC".equalsIgnoreCase(provider) || "BCEIDBUSINESS".equalsIgnoreCase(provider)) {
+      startTime = endTime.minus(configuration.getOtherSubmissionTimeWindow());
+      maxSubmissions = configuration.getOtherMaxSubmissions();
+    } 
+    else {
+        throw new IllegalArgumentException("Invalid provider " + provider);
```

Comment:
Avoid throwing exceptions directly when inside a code block/method that returns a reactive stream. Return a `Mono.error(new IllegalArgumentException(...))` instead.